compair/tests/algorithms/test_pair_adaptive_min_delta.py

```diff
@@ -0,0 +1,462 @@
+        self.assertEqual(max_key, 4)
+
+        ##
+        # Selects "randomly" if both criterion scores and scores are the same
+        scored_objects = [
+            ScoredObject(
+                key=1, score=0.9, variable1=None, variable2=None,
+                rounds=2, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=2, score=0.9, variable1=None, variable2=None,
+                rounds=2, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=3, score=0.9, variable1=None, variable2=None,
+                rounds=2, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=4, score=0.9, variable1=None, variable2=None,
+                rounds=1, wins=None, loses=None, opponents=None
+            )
+        ]
+        comparisons = []
+        criterion_scores = {
+            1: { 'c1': 11, 'c2': 21 },
+            2: { 'c1': 11, 'c2': 21 },
+            3: { 'c1': 11, 'c2': 21 },
+            4: { 'c1': 11, 'c2': 21 }
+        }
+        criterion_weights = {
+            'c1': 0.4, 'c2': 0.6
+        }
+
+        results = self.pair_algorithm.generate_pair(scored_objects, comparisons,
+            criterion_scores, criterion_weights)
+
+        # 4 should be selected first as it has lowest round.
+        # it should be paired with either 1, 2 or 3.
+        min_key = min([results.key1, results.key2])
+        max_key = max([results.key1, results.key2])
+        self.assertIn(min_key, [1, 2, 3])
+        self.assertEqual(max_key, 4)
+
+        ##
+        # Can select previously compared object but not with same opponent
+        scored_objects = [
+            ScoredObject(
+                key=1, score=0.5, variable1=None, variable2=None,
+                rounds=2, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=2, score=0.5, variable1=None, variable2=None,
+                rounds=2, wins=None, loses=None, opponents=None
+            ),
+            ScoredObject(
+                key=3, score=0.5, variable1=None, variable2=None,
+                rounds=3, wins=None, loses=None, opponents=None
+            )
+        ]
+        comparisons = [
+            ComparisonPair(1,2,None)
+        ]
+        criterion_scores = {
+            1: { 'c1': 10.1, 'c2': 20.2 },
+            2: { 'c1': 10.1, 'c2': 20.2 },
+            3: { 'c1': 10.1, 'c2': 20.2 }
+        }
+        criterion_weights = {
+            'c1': 0.4, 'c2': 0.6
+        }
+
+        results = self.pair_algorithm.generate_pair(scored_objects, comparisons,
+            criterion_scores, criterion_weights)
+
+        min_key = min([results.key1, results.key2])
+        max_key = max([results.key1, results.key2])
+        self.assertEqual(min_key, 1)
+        self.assertEqual(max_key, 3)
+
+        ##
+        # Select opponent with closest score (mock shuffle order)
```

Comment:
in this algorithm's tests, score is less important than criteria score so add a corresponding criteria score test that is in the same spirit as this one

Reply:
test case added
https://github.com/ubc/compair/blob/3837f3209f1394e878c852180b7667b9355cb69b/compair/tests/algorithms/test_pair_adaptive_min_delta.py#L430-L466